web/html/src/manager/virtualization/guests/list/guests-list.js

```diff
@@ -0,0 +1,418 @@
+const React = require("react");
+const PropTypes = React.PropTypes;
+const MessageContainer = require("../../../../components/messages").Messages;
+const MessagesUtils = require("../../../../components/messages").Utils;
+const {Table, Column, SearchField, Highlight} = require("../../../../components/table");
+const Network = require("../../../../utils/network");
+const Functions = require("../../../../utils/functions");
+const Utils = Functions.Utils;
+const {LinkButton, AsyncButton} = require("../../../../components/buttons");
+const {DeleteDialog} = require("../../../../components/dialog/DeleteDialog");
+const {DangerDialog} = require("../../../../components/dialog/DangerDialog");
+const {ModalButton} = require("../../../../components/dialog/ModalButton");
+const Systems = require("../../../../components/systems");
+import VirtualizationGuestActionApi from '../virtualization-guest-action-api';
+
+class GuestsList extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      actionsResults: {},
+      serverData: null,
+      errors: [],
+      selectedItems: []
+    }
+
+    this.refreshServerData = () => {
+      Network.get(`/rhn/manager/api/systems/details/virtualization/guests/${this.props.server_id}/data`, "application/json").promise
+        .then(data => {
+          this.setState({
+            serverData: data,
+            errors: [],
+          });
+        })
+        .catch(response => {
+          this.setState({
+            errors: response.status == 401 ? [t("Session expired, please reload the page to see up-to-date data.")] :
+              response.status >= 500 ? [t("Server error, please check log files.")] :
+              []
+          });
+        });
+    };
+
+    this.onBeforeUnload = (event) => {
+        this.setState({
+            pageUnloading: true
+        });
+    };
+
+    this.handleSelectItems = (items) => {
+      this.setState({
+        selectedItems: items
+      });
+    };
+
+    this.selectGuest = (row) => {
+      this.setState({
+        selected: row
+      });
+    };
+
+    this.action = (guests, type, value = "") => {
+      var messageData = {"uuids": guests, "value": value};
+      Network.post(`/rhn/manager/api/systems/details/virtualization/guests/${this.props.server_id}/${type}`,
```

Comment:
Cant't we reuse the VirtualizationGuestActionApi component here?

Reply:
I'ld go for another API component rather than forcing it into the actions one. Will add this component

Reply:
oh, sorry, reading PR comments while being in a conference isn't too good.
In fact this function can just be removed: it's no longer used with the introduction of the component you mentioned ;)

Reply:
that one is done, I just realized that I forgot to add the new file to the commit